lib/cube/buildCubeShape/index.js

```diff
@@ -92,15 +93,20 @@ class CubeShapeBuilder extends Transform {
 
   _flush (callback) {
     for (const cube of this.options.cubes.values()) {
-      this.push(cube.toDataset())
+      const dataset = cube.toDataset({ shapeGraph: toNamedNode(this.options.graph) })
+      this.push(dataset)
```

Comment:
I think this PR is unnecessarily complicated. I would revert changes in the `Cube` class and simply map the whole result to the selected graph

```suggestion
      const dataset = cube.toDataset()
      const graph = toNamedNode(this.options.graph) || $rdf.defaultGraph()

      this.push(dataset.map({ subject, predicate, object }) => 
        $rdf.quad(subject, predicate, object, graph))
```

Reply:
That was the initial implementation, 

The only problem is some triples belong to something other than a shape, and will end up with the shape the named-graph. 

For example:

<obs> a cube:ObservationSet .

Reply:
ah ok. the confusing part is that the module is called `buildCubeShape` although it does more than just cube. let me have another look